### PR TITLE
Implement native bindings for mount/umount

### PR DESCRIPTION
**Pull request by nwalters512**
This package implements native bindings for the `mount`/`umount` syscalls via a new `@prairielearn/bind-mount` package. This will be used in https://github.com/PrairieLearn/PrairieLearn/pull/1626. I pulled these changes out of that PR to minimize the amount of code that needs to be reviewed in that other PR.

### CODE REVIEW

File: packages/bind-mount/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if (err) {
+          reject(err);
+        } else {
+          resolve();
+        }
+      });
+    });
+  };
+
+  module.exports.unmount = (target) => {
```
Comment (mwest1066) [inline packages/bind-mount/index.js:16]: Not that I really mind, but would this be clearer as `umount()` to match the C lib function and the command line utility names?

**Review by mwest1066 — APPROVED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -2,6 +2,9 @@ FROM prairielearn/plbase
 
 ENV PATH="/PrairieLearn/node_modules/.bin:$PATH"
 
+# We need gcc-c++ to build the native code in `packages/bind-mount`.
+RUN yum install -y gcc-c++
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply Dockerfile:6]: Good call!

File: packages/bind-mount/index.js (hunk 1 above)
Comment (nwalters512) [reply packages/bind-mount/index.js:16]: That makes sense, yeah. Will change.

File: packages/bind-mount/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        }
+      });
+    });
+  };
+} catch (err) {
+  module.exports.mount = () => {
+    throw new Error('Failed to load native bindings');
+  };
+
+  module.exports.mount = () => {
```
[the remark this answers is not in the document]
Comment (nwalters512) [reply packages/bind-mount/index.js]: Yep, fixed in fda1caa where I renamed `unmount` to `umount`.

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (nwalters512) [reply Dockerfile]: Done in 4abc328.

[the remark this answers is not in the document]
Comment (mwest1066) [reply Dockerfile]: I believe I fixed the problem of the rebuilt `plbase` not being used in https://github.com/PrairieLearn/PrairieLearn/pull/5994/commits/2172d687848a671459de1ac14b1c6f2e628d82dc (part of #5994).

[the remark this answers is not in the document]
Comment (nwalters512) [reply Dockerfile]: Ah! Great find with that `--load` flag. `docker buildx` continues to be unintuitive to me 😂

**Pull request merged**